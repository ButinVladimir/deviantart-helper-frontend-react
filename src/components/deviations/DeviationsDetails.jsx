import React from 'react';
import DeviationsDetailsForm from './DeviationsDetailsForm';

export default function DeviationsDetails() {
  return (
    <div>
      <DeviationsDetailsForm />
    </div>
  );
}
