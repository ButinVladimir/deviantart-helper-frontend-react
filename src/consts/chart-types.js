export const CHART_VIEWS = 'views';
export const CHART_FAVOURITES = 'favourites';
export const CHART_COMMENTS = 'comments';
export const CHART_DOWNLOADS = 'downloads';
