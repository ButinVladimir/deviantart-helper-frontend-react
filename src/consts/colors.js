const LIGHT_BLUE = '#88f';
const RED = '#b33';
const GREEN = '#393';
const BLUE = '#11c';
const ORANGE = '#f94';
const PURPLE = '#94a';
const CYAN = '#2b9';
const BLACK = '#111';
const GREY = '#999';
const PINK = '#d5b';

export default [LIGHT_BLUE, RED, GREEN, BLUE, ORANGE, PURPLE, CYAN, BLACK, GREY, PINK];
