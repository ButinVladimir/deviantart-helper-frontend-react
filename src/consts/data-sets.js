export const DATA_SET_TIMESTAMP = 0;
export const DATA_SET_VIEWS = 1;
export const DATA_SET_FAVOURITES = 2;
export const DATA_SET_COMMENTS = 3;
export const DATA_SET_DOWNLOADS = 4;

export const dataSetOptions = [
  [DATA_SET_VIEWS, 'Views'],
  [DATA_SET_FAVOURITES, 'Favourites'],
  [DATA_SET_COMMENTS, 'Comments'],
  [DATA_SET_DOWNLOADS, 'Downloads'],
];
