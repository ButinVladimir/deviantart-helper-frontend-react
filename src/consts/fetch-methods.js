export const GET = 'GET';
export const POST = 'POST';
