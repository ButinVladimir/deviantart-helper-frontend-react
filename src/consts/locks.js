export const LOCK_LOAD_USER_INFO = 'LOCK_LOAD_USER_INFO';
export const LOCK_START_LOADING_DATA = 'LOCK_START_LOADING_DATA';
export const LOCK_REVOKE = 'LOCK_REVOKE';

export const LOCK_BROWSE_DEVIATIONS = 'LOCK_BROWSE_DEVIATIONS';
export const LOCK_DEVIATION_DETAILS = 'LOCK_DEVIATION_DETAILS';
export const LOCK_DEVIATION_DETAILS_METADATA = 'LOCK_DEVIATION_DETAILS_METADATA';
export const LOCK_DEVIATIONS_STATISTICS = 'LOCK_DEVIATIONS_STATISTICS';
export const LOCK_DEVIATIONS_TOTAL = 'LOCK_DEVIATIONS_TOTAL';
