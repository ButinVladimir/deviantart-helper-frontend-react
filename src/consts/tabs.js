export const DESCRIPTION = 'description';
export const PREVIEW = 'preview';
export const CHART = 'chart';
export const LIST = 'list';

export const deviationDetailsTabs = [
  [DESCRIPTION, 'Description'],
  [PREVIEW, 'Preview'],
  [CHART, 'Chart'],
];

export const deviationStatisticsTabs = [
  [LIST, 'List'],
  [CHART, 'Chart'],
];
