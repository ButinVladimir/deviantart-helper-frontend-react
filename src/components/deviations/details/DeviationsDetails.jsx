import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Section from 'react-bulma-components/lib/components/section';
import Container from 'react-bulma-components/lib/components/container';
import Content from 'react-bulma-components/lib/components/content';
import Heading from 'react-bulma-components/lib/components/heading';
import Tabs from 'react-bulma-components/lib/components/tabs';
import Level from 'react-bulma-components/lib/components/level';
import Loader from 'react-bulma-components/lib/components/loader';
import DeviationsDetailsDescriptionTab from './DeviationsDetailsDescriptionTabContainer';
import DeviationsDetailsPreviewTab from './DeviationsDetailsPreviewTabContainer';
import DeviationsDetailsChartTab from './DeviationsDetailsChartTabContainer';
import * as tabs from '../../../consts/tabs';
import convertTabs from '../../../helpers/convert-tabs';

export default class DeviationsDetails extends Component {
  componentDidMount() {
    const { clearDataHander, loadDeviationDetailsHandler } = this.props;

    clearDataHander();
    loadDeviationDetailsHandler();
  }

  render() {
    const {
      title,
      tab,
      detailsLoading,
      changeTabHandler,
    } = this.props;

    const mappedTabs = convertTabs(tabs.deviationDetailsTabs, tab, changeTabHandler);

    return (
      <>
        <Section>
          <Container>
            <Content>
              <Heading size={2}>{detailsLoading ? 'Deviation details' : title}</Heading>
            </Content>
            <Tabs type="toggle">
              {mappedTabs}
            </Tabs>
          </Container>
        </Section>

        {detailsLoading && (
          <Section>
            <Container>
              <Level>
                <Level.Item>
                  <Loader className="custom-loader" />
                </Level.Item>
              </Level>
            </Container>
          </Section>
        )}
        {!detailsLoading && (
          <>
            {tab === tabs.DESCRIPTION && <DeviationsDetailsDescriptionTab />}
            {tab === tabs.PREVIEW && <DeviationsDetailsPreviewTab />}
            {tab === tabs.CHART && <DeviationsDetailsChartTab />}
          </>
        )}
      </>
    );
  }
}

DeviationsDetails.propTypes = {
  title: PropTypes.string.isRequired,
  tab: PropTypes.string.isRequired,
  detailsLoading: PropTypes.bool.isRequired,
  clearDataHander: PropTypes.func.isRequired,
  changeTabHandler: PropTypes.func.isRequired,
  loadDeviationDetailsHandler: PropTypes.func.isRequired,
};
