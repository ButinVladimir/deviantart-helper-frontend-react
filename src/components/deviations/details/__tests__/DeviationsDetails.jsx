import React from 'react';
import { shallow } from 'enzyme';
import DeviationsDetails from '../DeviationsDetails';
import * as tabs from '../../../../consts/tabs';

describe('DeviationsDetails', () => {
  it('can be rendered correctly when details are loading', () => {
    const wrapper = shallow(
      <DeviationsDetails
        title="Details"
        tab={tabs.PREVIEW}
        detailsLoading
        clearDataHandler={() => {}}
        changeTabHandler={() => {}}
        loadDeviationDetailsHandler={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('when details are not loading', () => {
    const props = {
      title: 'Details',
      detailsLoading: false,
      clearDataHandler: () => {},
      changeTabHandler: () => {},
      loadDeviationDetailsHandler: () => {},
    };

    it('can be rendered when selected tab is description', () => {
      const wrapper = shallow(
        <DeviationsDetails
          {...props}
          tab={tabs.DESCRIPTION}
        />,
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('can be rendered when selected tab is preview', () => {
      const wrapper = shallow(
        <DeviationsDetails
          {...props}
          tab={tabs.PREVIEW}
        />,
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('can be rendered when selected tab is chart', () => {
      const wrapper = shallow(
        <DeviationsDetails
          {...props}
          tab={tabs.CHART}
        />,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('can handle clearing data and loading details', () => {
    const clearDataHandler = jest.fn();
    const loadDeviationDetailsHandler = jest.fn();
    shallow(
      <DeviationsDetails
        title="Details"
        tab={tabs.PREVIEW}
        detailsLoading
        clearDataHandler={clearDataHandler}
        changeTabHandler={() => {}}
        loadDeviationDetailsHandler={loadDeviationDetailsHandler}
      />,
    );

    expect(clearDataHandler).toHaveBeenCalled();
    expect(loadDeviationDetailsHandler).toHaveBeenCalled();
  });
});
