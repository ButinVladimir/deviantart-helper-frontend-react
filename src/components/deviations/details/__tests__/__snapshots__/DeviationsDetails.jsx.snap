// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsDetails can be rendered correctly when details are loading 1`] = `
<Fragment>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint={null}
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Content
        renderAs="div"
        size={null}
        style={Object {}}
      >
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={2}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          Deviation details
        </Heading>
      </Content>
      <Tabs
        align={null}
        className=""
        fullwidth={false}
        renderAs="div"
        size={null}
        style={Object {}}
        type="toggle"
      >
        <Tab
          active={false}
          className=""
          key="description"
          renderAs="a"
          style={Object {}}
        >
          Description
        </Tab>
        <Tab
          active={true}
          className=""
          key="preview"
          renderAs="a"
          style={Object {}}
        >
          Preview
        </Tab>
        <Tab
          active={false}
          className=""
          key="chart"
          renderAs="a"
          style={Object {}}
        >
          Chart
        </Tab>
      </Tabs>
    </Container>
  </Section>
  <CustomLoader />
</Fragment>
`;

exports[`DeviationsDetails when details are not loading can be rendered when selected tab is chart 1`] = `
<Fragment>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint={null}
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Content
        renderAs="div"
        size={null}
        style={Object {}}
      >
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={2}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          Details
        </Heading>
      </Content>
      <Tabs
        align={null}
        className=""
        fullwidth={false}
        renderAs="div"
        size={null}
        style={Object {}}
        type="toggle"
      >
        <Tab
          active={false}
          className=""
          key="description"
          renderAs="a"
          style={Object {}}
        >
          Description
        </Tab>
        <Tab
          active={false}
          className=""
          key="preview"
          renderAs="a"
          style={Object {}}
        >
          Preview
        </Tab>
        <Tab
          active={true}
          className=""
          key="chart"
          renderAs="a"
          style={Object {}}
        >
          Chart
        </Tab>
      </Tabs>
    </Container>
  </Section>
  <Component />
</Fragment>
`;

exports[`DeviationsDetails when details are not loading can be rendered when selected tab is description 1`] = `
<Fragment>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint={null}
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Content
        renderAs="div"
        size={null}
        style={Object {}}
      >
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={2}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          Details
        </Heading>
      </Content>
      <Tabs
        align={null}
        className=""
        fullwidth={false}
        renderAs="div"
        size={null}
        style={Object {}}
        type="toggle"
      >
        <Tab
          active={true}
          className=""
          key="description"
          renderAs="a"
          style={Object {}}
        >
          Description
        </Tab>
        <Tab
          active={false}
          className=""
          key="preview"
          renderAs="a"
          style={Object {}}
        >
          Preview
        </Tab>
        <Tab
          active={false}
          className=""
          key="chart"
          renderAs="a"
          style={Object {}}
        >
          Chart
        </Tab>
      </Tabs>
    </Container>
  </Section>
  <Connect(DeviationsDetailsDescriptionTab) />
</Fragment>
`;

exports[`DeviationsDetails when details are not loading can be rendered when selected tab is preview 1`] = `
<Fragment>
  <Section
    className=""
    renderAs="section"
    size={null}
    style={Object {}}
  >
    <Container
      breakpoint={null}
      fluid={false}
      renderAs="div"
      style={Object {}}
    >
      <Content
        renderAs="div"
        size={null}
        style={Object {}}
      >
        <Heading
          className=""
          heading={false}
          renderAs="h1"
          size={2}
          spaced={false}
          subtitle={false}
          weight={null}
        >
          Details
        </Heading>
      </Content>
      <Tabs
        align={null}
        className=""
        fullwidth={false}
        renderAs="div"
        size={null}
        style={Object {}}
        type="toggle"
      >
        <Tab
          active={false}
          className=""
          key="description"
          renderAs="a"
          style={Object {}}
        >
          Description
        </Tab>
        <Tab
          active={true}
          className=""
          key="preview"
          renderAs="a"
          style={Object {}}
        >
          Preview
        </Tab>
        <Tab
          active={false}
          className=""
          key="chart"
          renderAs="a"
          style={Object {}}
        >
          Chart
        </Tab>
      </Tabs>
    </Container>
  </Section>
  <Connect(DeviationsDetailsPreviewTab) />
</Fragment>
`;
